src/screens/RequestSwap/SwapScreen.tsx

```diff
@@ -1,109 +1,199 @@
+        (task) => !task.isSelected,
+    ];
 
-    //true if radio is checked or (if an index and a date are both selected)
-    let nextButtonClickable =
-        checked || (selectedIndex !== null && selectedSwapDate !== null);
+    //these operations only need to be run once on render. useEffect runs after everything is loaded so things that relied on tasks got angry at me
+    const tasks = React.useMemo(
+        () =>
+            swapCandidates
+                .reduce((acc, currentValue) => [...acc, ...currentValue.tasks], [])
+                .map((task) => ({ ...task, isSelected: false }))
+                .sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()), //may not need if swapCandidates are pre-sorted
+        [swapCandidates],
+    );
 
-    const onNextHandler = () => {
-        if (checked) {
-            dispatch(selectTargetTask(undefined));
-        } else {
-            dispatch(selectTargetTask(displayValue.tasks[0]));
-        }
-        props.navigation.navigate('SwapSummary');
+    const [filteredTasks, setFilteredTasks] = useState(tasks);
+    const selectedTasks = filteredTasks.filter((task) => task.isSelected);
+
+    //callbacks
+    const onItemSelect = (item, index) => {
+        setFilteredTasks((f) => {
+            let tempFilteredTasks = [...f];
+            tempFilteredTasks[index].isSelected = !tempFilteredTasks[index].isSelected;
+            return tempFilteredTasks;
+        });
+    };
+    const onTilePress = (date) => {
+        const dates = selectedDates.filter((item) => !compareDates(item, date));
+        dates.length === selectedDates.length
+            ? setSelectedDates([...selectedDates, date])
+            : setSelectedDates(dates);
     };
 
-    const candidates = swapCandidates.map((candidate, index) => {
+    //dropdown items for people select
+    const candidates = populateCandidates(swapCandidates);
+    const displayedPerson =
+        selectedPeopleIndices.length === 0
+            ? 'Name'
+            : selectedPeopleIndices.length === 1
+            ? `${swapCandidates[selectedPeopleIndices[0].row].firstName} ${
+                  swapCandidates[selectedPeopleIndices[0].row].lastName
+              }`
+            : `${swapCandidates[selectedPeopleIndices[0].row].firstName} ${
+                  swapCandidates[selectedPeopleIndices[0].row].lastName
+              }...`;
+
+    //dropdown items for time select
+    //TODO: hook up to actual task times
+    const possibleTimes = ['AM', 'PM']; //temp until database is finished
+    const times = populateTimes(possibleTimes);
+    const displayedTime =
+        selectedTimeIndices.length === 0
+            ? 'Time'
+            : selectedTimeIndices.length === 1
+            ? possibleTimes[selectedTimeIndices[0].row]
+            : `${possibleTimes[selectedTimeIndices[0].row]}...`;
+    //is there a better way to indicate that multiple times/people are selected besides "..." ? not too big a fan of how it looks
```

Comment:
A few ideas: 
(1) Say " and ..." instead:
```suggestion
            ? possibleTimes[selectedTimeIndices[0].row]
            : `${possibleTimes[selectedTimeIndices[0].row]} and ...`;
    //is there a better way to indicate that multiple times/people are selected besides "..." ? not too big a fan of how it looks
```
or
(2) Just use a hard-coded string: "(Multiple times selected)" or "(Multiple people selected)"
```suggestion
            ? possibleTimes[selectedTimeIndices[0].row]
            : '(Multiple times selected)';
    //is there a better way to indicate that multiple times/people are selected besides "..." ? not too big a fan of how it looks
```
or
(3) indicate a count: "(# times selected)"
```suggestion
            ? possibleTimes[selectedTimeIndices[0].row]
            : `(${selectedTimeIndices.length} times selected)`;
    //is there a better way to indicate that multiple times/people are selected besides "..." ? not too big a fan of how it looks
```

Reply:
ditto for [people selected](https://github.com/House-on-a-Rock/tjc-scheduling-app/pull/42/files/31674459ba40653688230aef3b4408898ad14863#r505156926)

Reply:
Hm i think i'll go with option 1 for now, there isn't really enough room to display much text.